src/components/Chart/ComparisonSelector.js

```diff
@@ -0,0 +1,191 @@
+/* eslint-disable consistent-return */
+/* eslint-disable no-restricted-syntax */
+import { useState, useContext, useEffect } from 'react';
+import {
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+} from '@mui/material';
+import axios from 'axios';
+import PropTypes from 'prop-types';
+import { filterSearch } from '../Common/Controller';
+import Notification from '../Common/Notification';
+import { DatastoreContext } from '../../context/DatastoreProvider';
+import env from '../../env';
+
+export const comparisonModeLabels = [
+  {
+    label: 'Default',
+    value: 'default',
+  },
+  {
+    label: 'Payors',
+    value: 'payors',
+    single: 'Payor',
+  },
+  {
+    label: 'Providers',
+    value: 'healthcareProviders',
+    single: 'Provider',
+  },
+  {
+    label: 'Coverage',
+    value: 'healthcareCoverages',
+    single: 'Coverage',
+  },
+  {
+    label: 'Practitioners',
+    value: 'healthcarePractitioners',
+    single: 'Practitioner',
+  },
+]
+
+function ComparisonSelector({
+  activeMeasure, handleResetData, isComposite, setIsLoading,
+  setDisplayData, setSelectedMeasures, setCurrentResults,
+}) {
+  const {
+    datastore: {
+      measurementYear, comparisonMode, filterOptions,
+    },
+    datastoreActions: {
+      setComparisonResults, setComparisonMode,
+    },
+  } = useContext(DatastoreContext);
+  const [error, setError] = useState(undefined);
+
+  const aliasObj = {
+    payors: 'Payors',
+    healthcareProviders: 'Providers',
+    healthcareCoverages: 'Coverage',
+    healthcarePractitioners: 'Practitioners',
+  };
+
+  // handles dailyMeasureResults in comparison mode no matter what,
+  // even if you were changing the year, went composite view to submeasure view,
+  // are in the member table view, anything that isn't expressly the selection dropdown
+  useEffect(() => {
+    if (!comparisonMode || comparisonMode === 'Default') {
+      return;
+    }
+
+    let didCancel = false;
+
+    const fetchFreshData = async () => {
+      setIsLoading(true);
+      const filterKey = Object.entries(aliasObj)
+        .find(([, label]) => label === comparisonMode)?.[0];
+      const items = filterKey ? (filterOptions[filterKey] || []) : [];
+      const allResults = [];
+
+      for (const item of items) {
+        // eslint-disable-next-line no-await-in-loop
+        const search = await filterSearch(
```

Comment:
Hmmmm, I'm unclear on what this `useEffect` is supposed to do. Seems out of place for the ComparisonSelector to handle filtering.

But also while testing I never ran into a situation where `items` had any values. It was always empty. Which means it never called `filterSearch` to begin with.

Do you know what situation this is supposed to handle? Because I've commented it out the entire useEffect on my end and works just fine.

Reply:
I was wrong, the member data works on @m-ox
[sonar cloud code smell fixes](https://github.com/amida-tech/saraswati-dashboard/pull/293/commits/3db3724edfc270cd256d4a5f4449f49e752bc554) with this useEffect turned off, and at some point, I got confused between commits thinking removing it was the problem. Turns out something broke along the way in the past 24 hours.

What I do know is the referenced commit does appear to work (before your commits hit), it just has ugly loading state, is also flat out ugly code and could use tidying, and the chart could use some better headers. How do we want to move forward? I could walk back to that commit and carefully fold in some of the newer work step by step...